src/org/omegat/core/team2/impl/GITCredentialsProvider.java

```diff
@@ -253,28 +209,37 @@ public boolean get(URIish uri, CredentialItem... items) throws UnsupportedCreden
                     throw new UnsupportedCredentialItem(uri, "Wrong fingerprint pattern");
                 }
                 if (predefinedFingerprint != null) {
-                    if (promptedFingerprint.equals(predefinedFingerprint)) {
-                        ((CredentialItem.YesNoType) i).setValue(true);
-                    } else {
-                        ((CredentialItem.YesNoType) i).setValue(false);
-                    }
+                    ((CredentialItem.YesNoType) i).setValue(promptedFingerprint.equals(predefinedFingerprint));
                     continue;
                 }
                 if (promptedFingerprint.equals(storedFingerprint)) {
                     ((CredentialItem.YesNoType) i).setValue(true);
                     continue;
                 }
-                int choice = Core.getMainWindow().showConfirmDialog(promptText, null,
-                        JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
-                if (choice == JOptionPane.YES_OPTION) {
+                IMainWindow mw = Core.getMainWindow();
+                if (mw != null) {  // on GUI application
+                    int choice = mw.showConfirmDialog(promptText, null,
+                            JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
+                    if (choice == JOptionPane.YES_OPTION) {
+                        ((CredentialItem.YesNoType) i).setValue(true);
+                        saveFingerprint(uri, promptedFingerprint);
+                    } else {
+                        ((CredentialItem.YesNoType) i).setValue(false);
+                    }
+                } else {
+                    // FIXME: implement me.
+                    // current silently agreed
                     ((CredentialItem.YesNoType) i).setValue(true);
```

Comment:
When console mode, it silently agreed a credential item. We should show credential finger print and ask yes/no.

Reply:
fixed with 68a2b0d

Reply:
Implemented.